backend/docker/event_handler.go

```diff
@@ -19,67 +19,20 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// Thread-safe collection of context.CancelFunc
-type cancellableCollection struct {
-	mutex sync.RWMutex
-	funcs map[string]context.CancelFunc
-}
-
-func newCancellableCollection() cancellableCollection {
-	return cancellableCollection{
-		mutex: sync.RWMutex{},
-		funcs: make(map[string]context.CancelFunc, 0),
-	}
-}
-
-func (c cancellableCollection) cancel(k string) bool {
-	c.mutex.Lock()
-	defer c.mutex.Unlock()
-
-	var f func()
-	var ok bool
-
-	if f, ok = c.funcs[k]; ok {
-		f()
-		delete(c.funcs, k)
-	}
-
-	return ok
-}
-
-func (c cancellableCollection) add(k string, ctx context.Context) context.Context {
-	c.mutex.Lock()
-	defer c.mutex.Unlock()
-
-	ctx, c.funcs[k] = context.WithCancel(ctx)
-	return ctx
-}
-
-func (c cancellableCollection) remove(k string) {
-	c.mutex.Lock()
-	defer c.mutex.Unlock()
-
-	if _, ok := c.funcs[k]; ok {
-		delete(c.funcs, k)
-	}
-}
-
-func (b Backend) ListenEventsForExported(ctx context.Context, promNetwork string) {
-	evtCh, errCh := b.cli.Events(ctx, types.EventsOptions{
+func (b DockerBackend) ListenForTasksToExport(ctx context.Context) {
```

Comment:
In the end this file would be replaced by a swarm event handler ?

Reply:
Not really, cause for Swarm, the `autoexporter` is gonna run with the docker backend and the `autoconfig` is gonna run with the swarm backend ;)